[docs] Add example to add a second icon next to the field's opening button (#12524)
mui/mui-x: docs/data/date-pickers/custom-opening-button/custom-opening-button.md

```

:::

> ## Add a second icon next to the opening button

Reviewer: Making it shorter 🙈 
It could also apply to the range picker and in that case, it wouldn't be the second icon, at least with the current implementation. 🤔 
```suggestion
## Add an icon next to the opening button
```

Author: Great point
I added a demo for the range picker and we can't use the `inputAdornment` there because we don't have it, instead we can use `textField.InputProps.endAdornment`

I'm not a fan of having 2 different DX depending on the component but I don't see how to solve this.
On the non-range picker I can't use `textField.InputProps.endAdornment` because we would loose the icon to open the picker AND the clearable icon.

Reviewer: This just shows that we clearly need more alignment in this regard and implementing the open button at least on Single Range Pickers. 🤔 
Anyway, that's a story for later time. 👍 

But on the other hand, WDYT, does it even make sense to have an `inputAdornment` slot?
If it can be done via `textField.InputProps.endAdornment`, why not try making it the only source of truth? 🤔

Author: One behavior change (I would not call that a breaking change) is that passing a `slots.inputAdornment` would now also impact mobile when using a responsive picker.
Maybe passing a variable in the owner state saying if we are on mobile or on desktop would be nice to allow to customize both independently.

Author: > If it can be done via textField.InputProps.endAdornment, why not try making it the only source of truth? 

Because, at least with the current implementation, you can access the children of the input adornment that we are passing, so you can't add something next to it without loosing the initial one.

I do agree that we need to clean this API, but it will probably be breaking in some way if we want to do it properly.

Maybe we can start by documenting what can be done with the current API, and create an issue to explore the best DX we can find to do all the customization around the adornments.

Reviewer: > Maybe passing a variable in the owner state saying if we are on mobile or on desktop would be nice to allow to customize both independently.

Great suggestion. 👍 
WDYT about firstly waiting for a validation if this is requested? Maybe before that we might have an open button mobile as well? 🙈 😆 

> Maybe we can start by documenting what can be done with the current API, and create an issue to explore the best DX we can find to do all the customization around the adornments.

Sure, it makes the most sense. 👌

> If you want to add a second icon next to the opening button, you can use the `inputAdornment` slot.

Reviewer: ```suggestion
If you want to add an icon next to the opening button, you can use the `inputAdornment` slot.
```
